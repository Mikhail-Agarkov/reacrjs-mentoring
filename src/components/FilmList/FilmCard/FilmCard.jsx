import React from 'react';

const FilmCard = (props) => (
    <div>
        <img src={props.src} alt={props.alt}/>
        <div>
            <span>{props.title}</span>
            <span>{props.date}</span>
        </div>
        <span>{props.description}</span>
    </div>
);

export default FilmCard;
