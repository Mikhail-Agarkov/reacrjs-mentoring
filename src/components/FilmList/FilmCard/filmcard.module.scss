.content {
  padding: 0 20px;
  width: 324px;
  height: 540px;
  font-family: Montserrat;
  color: #fff;
  cursor: pointer;
}

.cover {
  height: 455px;
  width: 324px;
}

.info {
  display: flex;
  justify-content: space-between;
  align-items: baseline;
  margin-top: 8px;
  font-size: 18px;
  opacity: 0.9;

  .date {
    border: 1px solid #979797;
    border-radius: 4px;
    padding: 2px 6px;
  }
}

.genre {
  font-size: 14px;
  opacity: 0.5;
}