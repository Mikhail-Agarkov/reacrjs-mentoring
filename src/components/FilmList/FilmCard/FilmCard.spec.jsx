import React from 'react';
import {shallow} from 'enzyme';

import FilmCard from "./FilmCard";

describe('FilmCard', () => {
    it('should render film card correctly', () => {
        const component = shallow(<FilmCard film={
            {
                'src': 'src/to/image',
                'alt': 'image alt text',
                'title': 'Film Title',
                'date': '2019',
                'genre': 'Fantasy',
                'key': '2516'
            }
        }/>);
        expect(component).toMatchSnapshot();
    })
});