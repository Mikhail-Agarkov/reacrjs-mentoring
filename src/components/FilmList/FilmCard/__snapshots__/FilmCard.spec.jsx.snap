// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard should render film card correctly 1`] = `
<div
  className="content"
  key="2516"
>
  <img
    alt="image alt text"
    className="cover"
    src="src/to/image"
  />
  <div
    className="info"
  >
    <span>
      Film Title
    </span>
    <span
      className="date"
    >
      2019
    </span>
  </div>
  <span
    className="genre"
  >
    Fantasy
  </span>
</div>
`;
